### PR TITLE
feat(static_obstacle_avoidance): force deactivation

### PR DESCRIPTION
**Pull request by go-sakayori**
## Description
Previously the static obstacle avoidance continuously activated even when the shift point was force deactivated by human operation via RTC. 
This PR enables force deactivation when yield maneuver could be applied, and after a certain time (this is parameterized) the avoidance would reactivate.

## Related links
Please merge [this PR](https://github.com/autowarefoundation/autoware_launch/pull/1101) first.

## How was this PR tested?
Before:
[Can't cancel](https://github.com/user-attachments/assets/53fb4361-9c8d-4b53-9f0e-6e00bbbfc9b1)
After:
[Duration time 2 second](https://github.com/user-attachments/assets/71f87f3e-0aa2-4904-83d2-349ed090b8d2)
[Duration time 10 second](https://github.com/user-attachments/assets/be528dad-42e8-4a6f-a4f4-483f0bbf03a0)

No degrade in [TIER IV internal scenario test](https://evaluation.tier4.jp/evaluation/reports/57af72fc-d3c4-52e3-88b3-587b54cdd9cd?project_id=prd_jt)

## Notes for reviewers
None.

## Interface changes
None.

## Effects on system behavior
None.

### CODE REVIEW

File: planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/src/scene.cpp [hunk 1]
```diff
@@ -570,6 +570,24 @@ void StaticObstacleAvoidanceModule::fillEgoStatus(
...
 
+  auto registered_sl_force_deactivated =
+    [&](const std::string & direction, const RegisteredShiftLineArray shift_line_array) {
+      return std::any_of(
+        shift_line_array.begin(), shift_line_array.end(), [&](const auto & shift_line) {
+          return rtc_interface_ptr_map_.at(direction)->isForceDeactivated(shift_line.uuid);
+        });
+    };
+
+  auto is_force_deactivated = registered_sl_force_deactivated("left", left_shift_array_) ||
```
Comment (satoshi-ota) [inline planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/src/scene.cpp:581]:
```suggestion
  const auto is_force_deactivated = registered_sl_force_deactivated("left", left_shift_array_) ||
```

File: planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/src/scene.cpp [hunk 2]
```diff
@@ -570,6 +570,24 @@ void StaticObstacleAvoidanceModule::fillEgoStatus(
     return;
   }
 
+  auto registered_sl_force_deactivated =
```
Comment (satoshi-ota) [inline planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/src/scene.cpp:573]:
```suggestion
  const auto registered_sl_force_deactivated =
```

**Review by satoshi-ota — APPROVED** [review]

File: planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/src/scene.cpp (hunk 1 above)
Comment (go-sakayori) [reply planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/src/scene.cpp]: Fixed in 22fa38d5b09c711d4e91948907a4f7af5bcbc193

File: planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/src/scene.cpp (hunk 2 above)
Comment (go-sakayori) [reply planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/src/scene.cpp]: Fixed in 22fa38d5b09c711d4e91948907a4f7af5bcbc193

**Review by satoshi-ota — APPROVED** [review]

**Pull request merged**